### ISSUE TITLE
Zen browser "forgets pins" when the last closed window isn't the one with the pins

### ISSUE THREAD

**Issue by l4kr** [zen-browser/desktop#2510]
### Captchas

- [X] I have read the instructions.
- [X] I have searched existing issues and avoided creating duplicates.
- [X] I am not filing an enhancement request.

### What happened?

When I drag a window out of the "main" A window - the new B window doesn't have the pins anymore.
I think this is fine and should stay like this. 

However, when I first close window A and then after that close window B - then reopen the browser the pins just aren't there anymore. 

Internally there should probably be a concept like the main and a secondary window.

### Reproducible?

- [X] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.1-a.16

### What platform are you seeing the problem on?

Windows

**Comment by lattaai1** [comment]
Hello, I've fixed this issue!

This is what I've done:
I've identified the issue with pins not persisting across browser restarts. The problem was that we weren't saving and restoring the pin-related attributes for tabs. I've updated the TabState.sys.mjs file to include these attributes when saving tab data, which should now persist pins across browser restarts.

I also created pull request: https://github.com/zen-browser/desktop/pull/2511


> [!CAUTION]  
> Disclaimer: This fix was created by Latta AI and you should never merge before you check the correctness of generated code!

---
This bug was fixed for free by Latta AI - https://latta.ai/ourmission
